src/vs/workbench/common/editor/editorGroupModel.ts

```diff
@@ -739,14 +742,14 @@ export class EditorGroupModel extends Disposable implements IEditorGroupModel {
 	private doSetSelection(newActiveEditor: EditorInput, activeEditorIndex: number, inactiveSelectedEditors: EditorInput[]): void {
 		this.doSetActive(newActiveEditor, activeEditorIndex);
 
-		for (const candidate of inactiveSelectedEditors) {
+		for (const candidate of inactiveSelectedEditors) { // TODO should this move up to the public API? the idea is that we do not trust editor inputs from public API but internal is fine
 			const editor = this.findEditor(candidate)?.[0];
 			if (editor && !this.isSelected(editor)) {
 				this.selection.push(editor);
 			}
 		}
```

Comment:
This is code from before. Why does it use `this.matches()`? Shouldn't comparing references be enough as they are taken directly from `this.editors`? I've moved to using `===` in most places for selection where the editor was taken from the model

Reply:
Yes, if we do compare with editors from the model, triple equals should be sufficient.